### ISSUE TITLE
"Use Native File Dialogs" has no effect

### ISSUE THREAD

**Issue by LuminousPath** [flathub/org.godotengine.Godot#187]
A new feature in Godot 4.3 is the ability to set whether you'd like to use the built-in file dialog for godot or use the OS native dialog. No matter what this option is set to it always defaults to the OS native dialog:
![image](https://github.com/user-attachments/assets/3dbb519d-e6c5-4bfe-bdab-b301745aa0de)

 (for me this is the GTK dialog):

![image](https://github.com/user-attachments/assets/201e66e7-f01f-42a1-8451-e9806c8fbef9)

While this wouldn't be a big deal on it's own, combined with [this bug](https://github.com/godotengine/godot/issues/92754) it makes flatpak godot 4.3 somewhat unusable for me right now.

Version: `v4.3.stable.flathub [77dcf97d8]`
OS: Guix System 6.9.12

**Comment by Spyder-0** [comment]
Yep, same issue here. It works on the non-flatpak version of Godot.
